addons/account/models/account_move.py

```diff
@@ -3291,7 +3292,7 @@ def _get_invoice_counterpart_amls_for_early_payment_discount(self, aml_values_li
                     # Track the balance to handle the exchange difference.
                     open_balance -= vals['balance']
 
-        exchange_diff_sign = aml.company_currency_id.compare_amounts(open_balance, 0.0)
+        exchange_diff_sign = aml and aml.company_currency_id.compare_amounts(open_balance, 0.0)
```

Comment:
doesn't seem like the right place to patch. You're supposed to receive valid elements in aml_values_list. Here you're patching in disregard of the method docstring :shrug: 

If that's the right patch, it should be made obvious, by 
1) adding a test case showcasing that the created journal entry correspond to the expactations
2) an update of the docstring as well

thanks :pray: